appservice/src/intTest/groovy/nu/yona/server/RemoveUserTest.groovy

```diff
@@ -51,61 +51,144 @@ class RemoveUserTest extends AbstractAppServiceIntegrationTest
+		disconnectMessage._links?."yona:process"?.href?.startsWith(getMessagesResponse.responseData._embedded."yona:messages".findAll{ it."@type" == "BuddyDisconnectMessage"}[0]._links.self.href)
+
 		def goalConflictMessages = getMessagesResponse.responseData._embedded."yona:messages".findAll{ it."@type" == "GoalConflictMessage"}
 		goalConflictMessages.size == 1
 		goalConflictMessages[0].nickname == "<self>"
 		goalConflictMessages[0]._links."yona:activityCategory".href == GAMBLING_ACT_CAT_URL
 		goalConflictMessages[0].url =~ /poker/
 		getMessagesResponse.responseData._embedded."yona:messages".findAll{ it."@type" == "BuddyConnectRequestMessage"}.size() == 0
 
+		def response = appService.postMessageActionWithPassword(buddyDisconnectMessages[0]._links."yona:process".href, [:], bob.password)
+		response.status == 200
+		response.responseData._embedded."yona:affectedMessages".size() == 1
+		response.responseData._embedded."yona:affectedMessages"[0]._links.self.href == disconnectMessage._links.self.href
+		response.responseData._embedded."yona:affectedMessages"[0]._links."yona:process" == null
+
 		cleanup:
 		appService.deleteUser(bob)
 	}
 
-	def 'Remove Richard and verify that buddy data is removed after Bob processed the remove buddy message'()
+	def 'Remove Richard with pending buddy request from Bob and verify that Bob receives a reject message'()
 	{
 		given:
-		def richardAndBob = addRichardAndBobAsBuddies()
-		def richard = richardAndBob.richard
-		def bob = richardAndBob.bob
-		analysisService.postToAnalysisEngine(bob, ["Gambling"], "http://www.poker.com")
-		analysisService.postToAnalysisEngine(richard, "news/media", "http://www.refdag.nl")
-		def message = "Goodbye friends! I deinstalled the Internet"
-		appService.deleteUser(richard, message)
-		def getResponse = appService.getMessages(bob)
-		def disconnectMessage = getResponse.responseData._embedded."yona:messages".findAll{ it."@type" == "BuddyDisconnectMessage"}[0]
-		def processURL = disconnectMessage._links."yona:process".href
+		def richard = addRichard()
+		def bob = addBob()
+		appService.sendBuddyConnectRequest(bob, richard)
 
 		when:
-		def response = appService.postMessageActionWithPassword(processURL, [:], bob.password)
+		def message = "Goodbye friends! I deinstalled the Internet"
+		appService.deleteUser(richard, message)
 
 		then:
+		def buddies = appService.getBuddies(bob)
+		buddies.size() == 0
+
+		def getMessagesResponse = appService.getMessages(bob)
+		getMessagesResponse.status == 200
+		getMessagesResponse.responseData._embedded
+		getMessagesResponse.responseData._embedded."yona:messages".size() == 1
+
+		def buddyConnectResponseMessages = getMessagesResponse.responseData._embedded."yona:messages".findAll{ it."@type" == "BuddyConnectResponseMessage"}
+		buddyConnectResponseMessages.size() == 1
+		def buddyConnectResponseMessage = buddyConnectResponseMessages[0]
+		buddyConnectResponseMessage.message == "User account was deleted"
```

Comment:
We could consider to add a `reason` here, such that the app can apply special style if required and because a user may fill the same message. But maybe that is not necessary.